cmd/srcd/daemon/daemon.go

```diff
@@ -42,8 +43,13 @@ func Client() (api.EngineClient, error) {
 	return api.NewEngineClient(conn), nil
 }
 
-func Start() error {
+func Start(workdir string) error {
+	return start(docker.WithEnv(workdirKey, workdir))
+}
+
+func start(opts ...docker.ConfigOption) error {
```

Comment:
I'd make start(opts) return a `func() error` so you don't need to pass the options around so much.

Closures ftw